### ISSUE TITLE
Issues with moving assets

### ISSUE THREAD

**Issue by cholmes** [stac-utils/stactools#146]
So I've been trying out my original use cases again to see if we truly need #145 

And I think I've hit two issues with moving assets that leave me a bit stuck. 

My use case is basically the merge described in the [stactools planet cli tutorial](https://stactools.readthedocs.io/en/latest/tutorials/working-with-planet-data-in-the-cli.html) - take two planet orders with the same capture but different assets and merge them into a single catalog, with items that have assets from both.

The first issue (which I encountered second, but is a bit easier to explain) is that a stac merge with moving assets doesn't seem to work with relative assets

```
(venv) cholmes@c02y3151jhd3 new-stacs % stac merge -a --ignore-conflicts planet-stac-sr/collection.json planet-stac/collection.json
Traceback (most recent call last):
  File "/Users/cholmes/Repos/planet-orders/venv/bin/stac", line 8, in <module>
    sys.exit(run_cli())
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/stactools/cli/cli.py", line 40, in run_cli
    cli(prog_name='stac')
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/stactools/cli/commands/merge.py", line 66, in merge_command
    child_folder=child_folder)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/stactools/cli/commands/merge.py", line 26, in merge
    child_folder)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/stactools/core/merge.py", line 113, in merge_all_items
    ignore_conflicts=ignore_conflicts)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/stactools/core/merge.py", line 44, in merge_items
    target_item, asset_href, ignore_conflicts=ignore_conflicts)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/stactools/core/copy.py", line 43, in move_asset_file_to_item
    raise ValueError('asset_href msut be absolute.')
ValueError: asset_href msut be absolute.
```

Ideally asset_href wouldn't need to absolute. But if it does then it'd be good to 1) document that in the reference docs, 2) mention it in the --help 3) raise it at the CLI level instead of blowing up a stack trace 4) fix the typo msut - must.

The second issue is harder for me to diagnose, due to #51, but if I don't do -a like above and then insstead do a move_assets after I do the merge then I get a no such file or directory error:

```
(venv) cholmes@c02y3151jhd3 new-stacs % stac merge --ignore-conflicts planet-stac-sr/collection.json planet-stac/collection.json    
(venv) cholmes@c02y3151jhd3 new-stacs % stac move-assets planet-stac/collection.json
Traceback (most recent call last):
  File "/Users/cholmes/Repos/planet-orders/venv/bin/stac", line 8, in <module>
    sys.exit(run_cli())
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/stactools/cli/cli.py", line 40, in run_cli
    cli(prog_name='stac')
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/stactools/cli/commands/copy.py", line 37, in move_assets_command
    copy=copy)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/stactools/core/copy.py", line 187, in move_all_assets
    for item in catalog.get_all_items():
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/pystac/catalog.py", line 437, in get_all_items
    yield from self.get_items()
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/pystac/stac_object.py", line 292, in get_stac_objects
    link.resolve_stac_object(root=self.get_root())
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/pystac/stac_object.py", line 202, in get_root
    root_link.resolve_stac_object()
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/pystac/link.py", line 213, in resolve_stac_object
    obj = stac_io.read_stac_object(target_href, root=root)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/pystac/stac_io.py", line 172, in read_stac_object
    d = self.read_json(source)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/pystac/stac_io.py", line 151, in read_json
    txt = self.read_text(source, *args, **kwargs)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/pystac/stac_io.py", line 215, in read_text
    return self.read_text_from_href(href, *args, **kwargs)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/stactools/core/io/__init__.py", line 24, in read_text_from_href
    with fsspec.open(href, "r") as f:
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/fsspec/core.py", line 102, in __enter__
    f = self.fs.open(self.path, mode=mode)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/fsspec/spec.py", line 968, in open
    **kwargs,
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/fsspec/implementations/local.py", line 132, in _open
    return LocalFileOpener(path, mode, fs=self, **kwargs)
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/fsspec/implementations/local.py", line 220, in __init__
    self._open()
  File "/Users/cholmes/Repos/planet-orders/venv/lib/python3.7/site-packages/fsspec/implementations/local.py", line 225, in _open
    self.f = open(self.path, mode=self.mode)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/cholmes/Repos/planet-orders/new-stacs/planet-stac/planet-stac/collection.json'
``` 

If I `cd` into planet-stac directory and run `stac move-assets collection.json` I get `No such file or directory: '/planet-stac/collection.json'`

It looks like the merge for some reason made the collection links like planet-stac/collection.json, which is weird. But then if I try a merge from within the planet-stac directory: ` stac merge --ignore-conflicts ../planet-stac-sr/collection.json collection.json` I get a `FileNotFoundError: [Errno 2] No such file or directory: '/collection.json'`

So sorry for a very unclean error report, but it seems like at least a couple things are off, and without #51 it's hard to know which file is even making the error.

**Comment by gadomski** [comment]
@cholmes I've opened #151, which adds a `stac validate` command. Right now the command is very naive, but it would be nice to make it robust enough for your needs and to fix #51. I've requested your review on #151, and if you have time to try out that command on your problematic catalogs, I can add powers to `stac validate` to hopefully give you the information you need to help debugging?

Alternatively, if there is an available catalog that I can play w/ that has similar issues to yours, I can play around as well and try to make `stac validate` a helpful tool.

**Comment by gadomski** [comment]
Closing as "likely fixed" per Gitter discussion with @cholmes, who will reopen/open new if need be.
